src/main/java/spoon/reflect/visitor/filter/SuperInheritanceHierarchyFunction.java

```diff
@@ -57,6 +57,47 @@
 	private boolean returnTypeReferences = false;
 
 	/**
+	 * Super inheritance hierarchy scanning listener.
+	 * Use it instead of {@link CtScannerListener}
+	 * if you need to know whether visited type reference is class or interface
+	 */
+	public static class Listener implements CtScannerListener {

```

Comment:
It is better `CtScannerListener`. so it might be interesting for some spoon clients. 
It is true that they might extend `DistinctTypeListener` too, but it provides some extra behavior, which might be NOT needed by some clients.
But technically I can make it private, so we keep Spoon API as small as possible.
So should I really make it private?